packages/edit-site/src/components/global-styles/screen-revisions/revisions-buttons.js

```diff
@@ -65,25 +72,40 @@ function getRevisionLabel( revision ) {
  * @return {JSX.Element} The modal component.
  */
 function RevisionsButtons( { userRevisions, selectedRevisionId, onChange } ) {
-	const currentTheme = useSelect(
-		( select ) => select( coreStore ).getCurrentTheme(),
-		[]
-	);
+	const { currentThemeName, currentUser } = useSelect( ( select ) => {
+		const { getCurrentTheme, getCurrentUser } = select( coreStore );
+		const currentTheme = getCurrentTheme();
+		return {
+			currentThemeName:
+				currentTheme?.name?.rendered || currentTheme?.stylesheet,
+			currentUser: getCurrentUser(),
+		};
+	}, [] );
+	const dateNowInMs = getDate().getTime();
+
 	return (
 		<ol
 			className="edit-site-global-styles-screen-revisions__revisions-list"
 			aria-label={ __( 'Global styles revisions' ) }
 			role="group"
 		>
 			{ userRevisions.map( ( revision, index ) => {
-				const { id, author, modified } = revision;
-				const authorDisplayName = author?.name || __( 'User' );
-				const authorAvatar = author?.avatar_urls?.[ '48' ];
-				const isUnsaved = 'unsaved' === revision?.id;
+				const { id, isLatest, author, modified } = revision;
+				const isUnsaved = 'unsaved' === id;
+				// Unsaved changes are created by the current user.
+				const revisionAuthor = isUnsaved ? currentUser : author;
+				const authorDisplayName = revisionAuthor?.name || __( 'User' );
+				const authorAvatar = revisionAuthor?.avatar_urls?.[ '48' ];
 				const isSelected = selectedRevisionId
-					? selectedRevisionId === revision?.id
+					? selectedRevisionId === id
 					: index === 0;
-				const isReset = 'parent' === revision?.id;
+				const isReset = 'parent' === id;
+				const modifiedDate = getDate( modified );
+				const displayDate =
+					modified &&
+					dateNowInMs - modifiedDate.getTime() > DAY_IN_MILLISECONDS
+						? dateI18n( dateFormat, modifiedDate )
```

Comment:
The date is formatted as **OCTOBER 20 2023**

This doesn't match the designs, which specify an abbreviated month — **OCT 20 2023** — however I think it's important to introduce new formats using the date library and into Core so that themes can override them.